packages/components/src/unit-control/test/utils.ts

```diff
@@ -42,6 +42,44 @@ describe( 'UnitControl utils', () => {
 			] );
 		} );
 
+		it( 'should not mutate default units argument definiton', () => {
+			const unitsA = useCustomUnits( {
+				availableUnits: [ 'px', 'em', 'rem' ],
+				defaultValues: { px: 8, em: 0.5, rem: 0.5 },
+			} );
+
+			const unitsB = useCustomUnits( {
+				availableUnits: [ 'px', 'em', 'rem' ],
+				defaultValues: { px: 16, em: 1, rem: 1 },
+			} );
+
+			expect( unitsA ).not.toEqual( unitsB );
+		} );
+
+		it( 'should not mutate custon units argument definitons', () => {
+			const unitsA = useCustomUnits( {
+				availableUnits: [ 'px', 'em', 'rem' ],
+				defaultValues: { px: 8, em: 0.5, rem: 0.5 },
+				units: [
+					{ value: 'px', label: 'pixel' },
+					{ value: 'em', label: 'em' },
+					{ value: 'rem', label: 'rem' },
+				],
+			} );
+
+			const unitsB = useCustomUnits( {
+				availableUnits: [ 'px', 'em', 'rem' ],
+				defaultValues: { px: 16, em: 1, rem: 1 },
+				units: [
+					{ value: 'px', label: 'pixel' },
+					{ value: 'em', label: 'em' },
+					{ value: 'rem', label: 'rem' },
+				],
+			} );
```

Comment:
```suggestion
			const units = [
				{ value: 'px', label: 'pixel' },
				{ value: 'em', label: 'em' },
				{ value: 'rem', label: 'rem' },
			];

			const unitsA = useCustomUnits( {
				availableUnits: [ 'px', 'em', 'rem' ],
				defaultValues: { px: 8, em: 0.5, rem: 0.5 },
				units,
			} );

			const unitsB = useCustomUnits( {
				availableUnits: [ 'px', 'em', 'rem' ],
				defaultValues: { px: 16, em: 1, rem: 1 },
				units,
			} );
```

Suggestion: Should we make this change to strictly verify that this unit test fails in the trunk branch and passes with this PR?